apps/admin/frontend/src/components/smartcard_modal/card_details_view.tsx

```diff
@@ -17,14 +22,17 @@ import {
 } from './status_message';
 import { userRoleToReadableString } from './user_roles';
 
-function checkDoesCardElectionHashMatchMachineElectionHash(
+function checkDoesCardElectionMatchMachineElection(
   programmedUser: User,
   electionDefinition: ElectionDefinition
 ): boolean {
-  if (!('electionHash' in programmedUser)) {
+  if (!('electionKey' in programmedUser)) {
```

Comment:
oh I guess this will supersede my previous comment and we will get invalid card instead of something went wrong?

Reply:
For programming purposes, the card will just show up as unprogrammed and allow you to program it. For auth purposes, the card will show up as an invalid card.

The full sequence of events will be as follows:

- Card cert parsing will error because of an incompatibility with [this schema](https://github.com/votingworks/vxsuite/blob/6eb372e8874825c60dcbb157f0b5ff08a7e90f38/libs/auth/src/certs.ts#L157-L168)
- The `JavaCard` layer [will return `undefined` card details](https://github.com/votingworks/vxsuite/blob/6eb372e8874825c60dcbb157f0b5ff08a7e90f38/libs/auth/src/java_card.ts#L445)
- The auth layer will return `invalid_user_on_card` if you try to auth with the card ([link]( https://github.com/votingworks/vxsuite/blob/6eb372e8874825c60dcbb157f0b5ff08a7e90f38/libs/auth/src/dipped_smart_card_auth.ts#L626-L628)) or that the card is unprogrammed in programming contexts ([link](https://github.com/votingworks/vxsuite/blob/6eb372e8874825c60dcbb157f0b5ff08a7e90f38/libs/auth/src/dipped_smart_card_auth.ts#L61-L80))

Reply:
This condition is really just detecting sys admin cards, which have no election key. The backend will never expose an old-style election manager or poll worker user to the frontend